omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/DatabaseChangeResource2_0.java

```diff
@@ -0,0 +1,167 @@
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			return description;
+		} else if (rep instanceof FullRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty(UUID);
+			description.addProperty(DISPLAY);
+			description.addProperty(AUTHOR);
+			description.addProperty(DESCRIPTION);
+			description.addProperty(RUN_STATUS);
+			description.addProperty(COMMENTS);
+			description.addProperty(RAN_DATE);
+			description.addSelfLink();
+			return description;
+		} else if (rep instanceof RefRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty(UUID);
+			description.addProperty(DISPLAY);
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			return description;
+		}
+		return null;
+	}
+
+	@Override
+	public PageableResult doGetAll(RequestContext context) {
+		return new NeedsPaging<>(getDatabaseChanges(), context);
+	}
+
+	@Override
+	public DatabaseUpdater.OpenMRSChangeSet getByUniqueId(String uniqueId) {
+		DatabaseUpdater.OpenMRSChangeSet databaseChange = getDatabaseChangeById(uniqueId);
+
+		if (databaseChange == null) {
+			throw new ObjectNotFoundException("Database change with id: " + uniqueId + " doesn't exist.");
+		}
+
+		return databaseChange;
+	}
+
+	@PropertyGetter(UUID)
+	public static String getUuid(DatabaseUpdater.OpenMRSChangeSet instance) {
+		return instance.getId();
+	}
+
+	@PropertyGetter(DISPLAY)
+	public static String getDisplay(DatabaseUpdater.OpenMRSChangeSet instance) {
+		return instance.getAuthor() + " " + instance.getDescription();
+	}
+
+	@Override
+	public Model getGETModel(Representation rep) {
+		ModelImpl model = (ModelImpl) super.getGETModel(rep);
+		if (rep instanceof DefaultRepresentation) {
+			model
+					.property(UUID, new StringProperty())
+					.property(DISPLAY, new StringProperty())
+					.property(AUTHOR, new StringProperty())
+					.property(DESCRIPTION, new StringProperty())
+					.property(RUN_STATUS, new EnumProperty(ChangeSet.RunStatus.class));
+		} else if (rep instanceof FullRepresentation) {
+			model
+					.property(UUID, new StringProperty())
+					.property(DISPLAY, new StringProperty())
+					.property(AUTHOR, new StringProperty())
+					.property(DESCRIPTION, new StringProperty())
+					.property(RUN_STATUS, new EnumProperty(ChangeSet.RunStatus.class))
+					.property(COMMENTS, new StringProperty())
+					.property(RAN_DATE, new DateProperty());
+		} else if (rep instanceof RefRepresentation) {
+			model
+					.property(UUID, new StringProperty())
+					.property(DISPLAY, new StringProperty());
+		}
+		return model;
+	}
+
+	@Override
+	public DatabaseUpdater.OpenMRSChangeSet newDelegate() {
+		return null;
+	}
+
+	private List<DatabaseUpdater.OpenMRSChangeSet> getDatabaseChanges() {
+		List<DatabaseUpdater.OpenMRSChangeSet> databaseChanges;
+		try {
```

Comment:
What happens if we do not catch this exception?

Reply:
It's a checked exception so either we wrap this call in try-catch block or add it to the method signature, that this method throws. If we add that to the method signature, then we would have to add the same `throws Exception` to `getByUniqueId` and `doGetAll`, which is illegal, as they don't throw anything in the interface.